src/net.cpp

```diff
@@ -1031,6 +1060,9 @@ void ThreadSocketHandler()
             BOOST_FOREACH(CNode* pnode, vNodesCopy)
                 pnode->Release();
         }
+
+        if (progress == 0)  // Nothing happened even though select did not block.  So slow us down.
+            MilliSleep(50);
```

Comment:
perhaps you could turn off the 50 millisecond sleep when rate limiting
is not turned on so that it doesn't impact the performance for those
wanting to rate limit...

On 15/10/2015 6:23 AM, gandrewstone wrote:
>
> In src/net.cpp
> <https://github.com/bitcoinxt/bitcoinxt/pull/81#discussion_r42122953>:
>
> > @@ -1031,6 +1060,9 @@ void ThreadSocketHandler()
> >              BOOST_FOREACH(CNode* pnode, vNodesCopy)
> >                  pnode->Release();
> >          }
> > +
> > +        if (progress == 0)  // Nothing happened even though select did not block.  So slow us down.
> > +            MilliSleep(50);
>
> Select won't block because there is data available. But we are not
> "harvesting" it because there is not enough room in the leaky bucket.
>
> 50msec is unfortunately completely arbitrary.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bitcoinxt/bitcoinxt/pull/81/files#r42122953>.
>